odl/operator/tensor_ops.py

```diff
@@ -866,8 +868,14 @@ def _call(self, x, out=None):
                 # sparse matrices
                 out[:] = self.matrix.dot(x)
             else:
-                with writable_array(out) as out_arr:
-                    self.matrix.dot(x, out=out_arr)
+                if (parse_version(np.__version__) < parse_version('1.13.0') and
+                        x is out):
```

Comment:
This does not catch all cases, e.g. we could have to containers that are possibly alliased, overall this is a rather hard problem to solve, see e.g.

https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.shares_memory.html
https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.may_share_memory.html

but I guess this solves some of the cases?

Reply:
True, this isn't a perfect fix, but for our current classes it should handle everything. In particular since we don't support `MatrixOperator` on product spaces (where things can become hairy).